From: Debian Nginx Maintainers <pkg-nginx-maintainers@lists.alioth.debian.org>
Date: Sun, 23 Oct 2016 19:46:14 -0700
Subject: perl-use-dpkg-buildflags
Description: Use linker flags from environment for perl (dpkg-buildflags).
 Necessary for hardening flags.
Author: Christos Trochalakis <yatiohi@ideopolis.gr>

---
 src/http/modules/perl/Makefile.PL | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/http/modules/perl/Makefile.PL b/src/http/modules/perl/Makefile.PL
index 03348b5..7d544ae 100644
--- a/src/http/modules/perl/Makefile.PL
+++ b/src/http/modules/perl/Makefile.PL
@@ -3,6 +3,7 @@
 # Copyright (C) Nginx, Inc.
 
 use 5.006001;
+use Config;
 use ExtUtils::MakeMaker;
 
 WriteMakefile(
@@ -14,6 +15,9 @@ WriteMakefile(
     AUTHOR            => 'Igor Sysoev',
 
     CCFLAGS           => "$ENV{NGX_PM_CFLAGS}",
+    # Pass link hardening flags
+    # $Config{lddlflags} is the default
+    LDDLFLAGS         => "$Config{lddlflags} $ENV{DEBIAN_NGINX_PERL_LDFLAGS}",
     OPTIMIZE          => '-O',
 
     INC               => join(" ", map {
